src/platform/datapath_winuser.c

```diff
@@ -3877,13 +3879,15 @@ CxPlatSocketSendInline(
             PktInfo6->ipi6_addr = LocalAddress->Ipv6.sin6_addr;
         }
 
-        WSAMhdr.Control.len += WSA_CMSG_SPACE(sizeof(INT));
-        CMsg = WSA_CMSG_NXTHDR(&WSAMhdr, CMsg);
-        CXPLAT_DBG_ASSERT(CMsg != NULL);
-        CMsg->cmsg_level = IPPROTO_IPV6;
-        CMsg->cmsg_type = IPV6_ECN;
-        CMsg->cmsg_len = WSA_CMSG_LEN(sizeof(INT));
-        *(PINT)WSA_CMSG_DATA(CMsg) = SendData->ECN;
+        if (SendData->ECN != CXPLAT_ECN_NON_ECT) {
```

Comment:
It would be less duplicated code and cleaner if we move it out of the PKT_INFO block.
```
        if (SendData->ECN != CXPLAT_ECN_NON_ECT) {
            WSAMhdr.Control.len += WSA_CMSG_SPACE(sizeof(INT));
            CMsg = WSA_CMSG_NXTHDR(&WSAMhdr, CMsg);
            CXPLAT_DBG_ASSERT(CMsg != NULL);
            CMsg->cmsg_level = LocalAddress->si_family == QUIC_ADDRESS_FAMILY_INET ?
                IPPROTO_IP : IPPROTO_IPV6;
            CMsg->cmsg_type = IP_ECN; // == IPV6_ECN
            CMsg->cmsg_len = WSA_CMSG_LEN(sizeof(INT));
            *(PINT)WSA_CMSG_DATA(CMsg) = SendData->ECN;
        } 
```

Reply:
I prefer to leave as is, since it's fewer branches.

Reply:
There actually won't be any additional branches if you use IP_ECN for both v4 and v6 and use a variable to store cmsg_level for ECN and have it set in the PKT_INFO block. Up to you.